gtk4/src/subclass/widget.rs

```diff
@@ -895,6 +967,43 @@ pub unsafe trait WidgetClassSubclassExt: ClassStruct {
         }
     }
 
+    fn install_property_action(&mut self, action_name: &str, property_name: &str) {
+        unsafe {
+            let widget_class = self as *mut _ as *mut ffi::GtkWidgetClass;
+            ffi::gtk_widget_class_install_property_action(
+                widget_class,
+                action_name.to_glib_none().0,
+                property_name.to_glib_none().0,
+            );
+        }
+    }
+
+    fn get_activate_signal(&self) -> SignalId {
+        unsafe {
+            let widget_class = self as *mut _ as *mut ffi::GtkWidgetClass;
+            let signal_id = ffi::gtk_widget_class_get_activate_signal(widget_class);
+            let type_ = Self::Type;
+            SignalId(type_, signal_id)
```

Comment:
We can't construct the SignalId as the constructor is private, I wonder if it makes sense to make it public?

Reply:
Yes, feel free to add a public, unsafe constructor for this. It probably also makes sense to have public safe variants of `g_signal_query`, `g_signal_lookup`, `g_signal_name`.

Also the type is only really in the `SignalId` for checking during `emit`. The signal ids are unique per process, not just unique per type.